pandas/tests/series/test_constructors.py

```diff
@@ -725,3 +725,14 @@ def f():
         self.assertEqual(s.dtype, 'timedelta64[ns]')
         s = Series([pd.NaT, np.nan, '1 Day'])
         self.assertEqual(s.dtype, 'timedelta64[ns]')
+
+    def test_constructor_name_hashable(self):
+        for n in [777, 777., 'name', datetime(2001, 11, 11), (1, 2)]:
+            for data in [[1, 2, 3], np.ones(3), {'a': 0, 'b': 1}]:
+                s = Series(data, name=n)
+                self.assertEqual(s.name, n)
```

Comment:
need to add similar tests for setting ``.name`` (search for where these existing tests are). 

e.g.

```
s = Series(data)
s.name = n
```
type things